packages/vue-app/template/store.js

```diff
@@ -3,130 +3,136 @@ import Vuex from 'vuex'
   return new Vuex.Store(Object.assign({
     strict: (process.env.NODE_ENV !== 'production')
-  }, storeData, {
-    state: storeData.state instanceof Function ? storeData.state() : {}
-  }))
+  }, store))
 }
 
 // Dynamically require module
-function getModule(filename) {
-  const file = files(filename)
-  const module = file.default || file
-  if (module.commit) {
-    throw new Error('[nuxt] <%= dir.store %>/' + filename.replace('./', '') + ' should export a method which returns a Vuex instance.')
+function requireModule(path, { isRoot = false, isState = false } = {}) {
+  const file = fileResolver(path)
+  const moduleData = file.default || file
+
+  if (isState && typeof moduleData !== 'function') {
+    return () => moduleData
   }
-  if (module.state && typeof module.state !== 'function') {
-    throw new Error('[nuxt] state should be a function in <%= dir.store %>/' + filename.replace('./', ''))
+  if (isRoot && moduleData.commit) {
+    throw new Error('[nuxt] <%= dir.store %>/' + path.replace('./', '') + ' should export a method which returns a Vuex instance.')
   }
-  return module
+
+  if (isRoot) {
+    // Avoid TypeError: setting a property that has only a getter when overwriting top level keys
+    const state = moduleData.state && moduleData.state !== 'function' ? (() => state) : moduleData.state
+    return Object.assign({}, moduleData, { state })
+  }
+  return moduleData
 }
 
-function getModuleNamespace(storeData, namePath, forAppend = false) {
-  if (namePath.length === 1) {
-    if (forAppend) {
-      return storeData
-    }
-    return storeData.modules
+function getStoreModule(storeModule, namespaces, { isProperty = false } = {}) {
+  // If ./mutations.js
+  if (namespaces.length === 1 && isProperty) {
+    return storeModule
+  }
+  if (namespaces.length === 0) {
+    return storeModule.modules
   }
-  const namespace = namePath.shift()
-  storeData.modules[namespace] = storeData.modules[namespace] || {}
-  storeData.modules[namespace].namespaced = true
-  storeData.modules[namespace].modules = storeData.modules[namespace].modules || {}
-  return getModuleNamespace(storeData.modules[namespace], namePath, forAppend)
+
+  const namespace = namespaces.shift()
+
+  storeModule.modules[namespace] = storeModule.modules[namespace] || {}
+  storeModule.modules[namespace].namespaced = true
+  storeModule.modules[namespace].modules = storeModule.modules[namespace].modules || {}
+
+  return getStoreModule(storeModule.modules[namespace], namespaces, { isProperty })
 }
 
-function appendModule(module, filename, name) {
-  const file = files(filename)
-  module.appends = module.appends || []
-  module.appends.push(name)
-  module[name] = file.default || file
+function mergeProperty(storeModule, moduleData, property) {
+  if (!moduleData) return
+
+  if (property === 'state') {
+    storeModule.state = moduleData || storeModule.state
+  } else {
+    storeModule[property] = Object.assign({}, storeModule[property], moduleData)
```

Comment:
```suggestion
    Object.assign(storeModule[property], moduleData)
```

Maybe? :relaxed:

Reply:
I don’t think it will work since it will replace the whole content of the property